includes/modules/order_total/ot_coupon.php

```diff
@@ -162,6 +162,11 @@ function collect_posts() {
     if (isset($_POST['dc_redeem_code']) && strtoupper($_POST['dc_redeem_code']) == 'REMOVE') {
       unset($_POST['dc_redeem_code']);
       unset($_SESSION['cc_id']);
+      
+      $GLOBALS['zco_notifier']->notify(
+          'NOTIFY_OT_COUPON_COUPON_REMOVED'
+       );
+       

```

Comment:
The coupon's either registered (in which case, the information was sent via a previous notification) or not, since there's only one coupon-per-order.  _This_ notification serves to allow an observer to stop "remembering" any coupon previously registered for the order.

Reply:
k, makes sense